src/backend/distributed/commands/multi_copy.c

```diff
@@ -3184,6 +3316,17 @@ InitializeCopyShardState(CopyShardState *shardState,
 }
 
 
+/*
+ * LogLocalCopyExecution logs that the copy will be done locally for
+ * the given shard.
+ */
+static void
+LogLocalCopyExecution(uint64 shardId)
+{
+	ereport(NOTICE, (errmsg("executing the copy locally for shard")));
```

Comment:
Should we log the shardid here?

Reply:
why do we have a separate `LogLocal<Foo>` function ?

Reply:
This is a separate code path, we dont use local execution for doing local copy. I thought of maybe including the shard id in the log. But if we don't want to include it we can make LogLocalCommand public and use that with some changes ( as it takes a task as an argument unnecessarily)

Reply:
would be useful to at least print the shard ID